packages/flutter/lib/src/services/raw_keyboard_android.dart

```diff
@@ -0,0 +1,205 @@
+  /// The hardware scan code corresponding to this key event.
+  ///
+  /// See <https://developer.android.com/reference/android/view/KeyEvent.html#getScanCode()>
+  /// for more information.
+  final int scanCode;
+
+  /// The modifiers that were present when the key event occurred.
+  ///
+  /// See <https://developer.android.com/reference/android/view/KeyEvent.html#getMetaState()>
+  /// for the numerical values of the `metaState`. Many of these constants are
+  /// also replicated as static constants in this class.
+  ///
+  /// See also:
+  ///
+  ///  * The [modifiersPressed] accessor to get a Set of currently pressed
+  ///    modifiers.
+  ///  * The [isModifierPressed] function to query if a specific modifier is
+  ///    pressed.
+  ///  * The [isControlPressed] convenience accessor to see if the CTRL key is
+  ///    pressed.
+  ///  * The [isShiftPressed] convenience accessor to see if the SHIFT key is
+  ///    pressed.
+  ///  * The [isAltPressed] convenience accessor to see if the ALT key is
+  ///    pressed.
+  ///  * The [isMetaPressed] convenience accessor to see if the META key is
+  ///    pressed.
+  final int metaState;
+
+  bool _isLeftRightModifierPressed(KeyboardSide side, int anyMask, int leftMask, int rightMask) {
+    if (metaState & anyMask == 0) {
+      return false;
+    }
+    switch (side) {
+      case KeyboardSide.any:
+        return true;
+      case KeyboardSide.both:
+        return metaState & leftMask != 0 && metaState & rightMask != 0;
+      case KeyboardSide.left:
+        return metaState & leftMask != 0;
+      case KeyboardSide.right:
+        return metaState & rightMask != 0;
+    }
+    return false;
+  }
+
+  @override
+  bool isModifierPressed(ModifierKey key, {KeyboardSide side = KeyboardSide.any}) {
+    assert(side != null);
+    switch (key) {
+      case ModifierKey.controlModifier:
+        return _isLeftRightModifierPressed(side, modifierControl, modifierLeftControl, modifierRightControl);
+      case ModifierKey.shiftModifier:
+        return _isLeftRightModifierPressed(side, modifierShift, modifierLeftShift, modifierRightShift);
+      case ModifierKey.altModifier:
+        return _isLeftRightModifierPressed(side, modifierAlt, modifierLeftAlt, modifierRightAlt);
+      case ModifierKey.metaModifier:
+        return _isLeftRightModifierPressed(side, modifierMeta, modifierLeftMeta, modifierRightMeta);
+      case ModifierKey.capsLockModifier:
+        return metaState & modifierCapsLock != 0;
+      case ModifierKey.numLockModifier:
+        return metaState & modifierNumLock != 0;
+      case ModifierKey.scrollLockModifier:
+        return metaState & modifierScrollLock != 0;
+      case ModifierKey.functionModifier:
+        return metaState & modifierFunction != 0;
+      case ModifierKey.symbolModifier:
+        return metaState & modifierSym != 0;
+    }
+    return false;
+  }
+
+  // Modifier key masks.
+
+  /// No modifier keys are pressed.
+  static const int modifierNone = 0;
```

Comment:
Is there value in making these all part of the public interface? If yes, should these include a "see also" to link people to the more convenient methods of determining these things?

Reply:
There is marginal value: we expose the `metaState` variable, so if you actually want to know what state is represented there, these masks will tell you.  Added a sentence to each of these referring tot he "easier" way.